api/src/main/java/org/pmiops/workbench/config/RetryConfig.java

```diff
@@ -0,0 +1,84 @@
+package org.pmiops.workbench.config;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.pmiops.workbench.exceptions.ExceptionUtils;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.backoff.BackOffPolicy;
+import org.springframework.retry.backoff.ExponentialRandomBackOffPolicy;
+import org.springframework.retry.backoff.Sleeper;
+import org.springframework.retry.backoff.ThreadWaitSleeper;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+
+@Configuration
+public class RetryConfig {
+
+  public static abstract class ResponseCodeRetryPolicy
+      extends SimpleRetryPolicy {
+
+    private static final Logger logger = Logger.getLogger(ResponseCodeRetryPolicy.class.getName());
+
+    private final String serviceName;
+
+    public ResponseCodeRetryPolicy(String serviceName) {
+      this.serviceName = serviceName;
+    }
+
+    @Override
+    public boolean canRetry(RetryContext context) {
+      if (context.getLastThrowable() == null) {
+        return true;
```

Comment:
Could you add a comment here? Not sure I follow why no exception ---> canRetry

Reply:
Added a comment. Weirdly, canRetry() gets called before the first call, and you're supposed to return true there if you want it to do anything. :)